### PR TITLE
Fix the buffer as primary key merge bug

### PR DESCRIPTION
**Pull request by sevenryze**
See the #4220 for details.

I guess this is my third PR which to resolve the Buffer kind of issues. LOL

You may give me the buffer sufferer reward. 😄

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ If we missed a note on some change or you have a questions on migrating from old
 feel free to ask us and community.
 
 ## 0.2.18 (UNRELEASED)
+1. fixed the getMany() result being droped randomly bug when using the buffer as primary key. ([#4220](https://github.com/typeorm/typeorm/issues/4220))
```
Comment (rustamwin) [inline CHANGELOG.md]: This line should be in `Bug fixes` section.

**Comment by pleerock** [comment]
You might be the only one who is using buffer as primary key 😆

**Comment by sevenryze** [comment]
@pleerock Hahahaha, it proves that typeORM is the best ORM in terms of nodejs ecosystem which could even support wired cases like this one. 😄 

Anyway, Really fast response as you, thanks for your review and quick fixing !

BTW, could you please give me the rough timeline when this fix is gonna live? I need to use it recently and it's cumbersome to replace the nodejs resolving function to my temp workaround version, especially for CI/CD environment. Thank you again.

File: CHANGELOG.md (hunk 1 above)
Comment (sevenryze) [reply CHANGELOG.md]: Sorry for that, i might misunderstand the contrib guide, which is saying I need to place this line inside the section unreleased, So i only searched for unreleased literally and placed this line below it.. LOL

**Comment by pleerock** [comment]
I'll release a new version on Monday or Tuesday

**Pull request merged**